arroyo-types/src/api.rs

```diff
@@ -0,0 +1,74 @@
+use serde::{Deserialize, Serialize};
+use utoipa::ToSchema;
+
+#[derive(Serialize, Deserialize, Clone, ToSchema, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct PipelinePost {
+    pub name: String,
+    pub definition: Option<String>,
+    pub udfs: Option<Vec<Udf>>,
```

Comment:
I think this can just be `Vec<Udf>` and the client can pass an empty array if they're not defining any; generally you'd only want to wrap Vec in Option if there's an semantic difference between the empty list and not having it be set.